max-statements-per-line and es6 export

**What version of ESLint are you using?**
v2.11.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Default (babel-eslint shows the same issue)

**Please show your full configuration:**
```json
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "rules": {
    "max-statements-per-line": [1, {"max": 1}]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
export default function someFunction() {
    return true;
}
```

**What did you expect to happen?**
No warnings.

**What actually happened? Please include the actual, raw output from ESLint.**
```

/path-to-the-file/index.js
  1:16  warning  This line has too many statements. Maximum allowed is 1  max-statements-per-line

✖ 1 problem (0 errors, 1 warning)

```

So… Maybe we could say that I have **two** statements because I'm declaring a function and exporting it. But in my opinion that is not the case, it's a special one related to ES6 syntax.

But, if we really consider that there are two statements in that line, it would be nice to have an option for `max-statements-per-line` rule that we could ignore cases like this.

--- a/lib/rules/max-statements-per-line.js
+++ b/lib/rules/max-statements-per-line.js
@@ -43,7 +43,7 @@ module.exports = {
         // Helpers
         //--------------------------------------------------------------------------
 
-        var SINGLE_CHILD_ALLOWED = /^(?:DoWhile|For|ForIn|ForOf|If|Labeled|While)Statement$/;
+        var SINGLE_CHILD_ALLOWED = /^(?:(?:DoWhile|For|ForIn|ForOf|If|Labeled|While)Statement|Export(?:Default|Named)Declaration)$/;
 
         /**
          * Gets the actual last token of a given node.

--- a/tests/lib/rules/max-statements-per-line.js
+++ b/tests/lib/rules/max-statements-per-line.js
@@ -86,6 +86,34 @@ ruleTester.run("max-statements-per-line", rule, {
                 "else",
                 "    foo++;"
             ].join("\n")
+        },
+        {
+            code: "export default foo = 0;",
+            options: [{max: 1}],
+            parserOptions: {ecmaVersion: 6, sourceType: "module"}
+        },
+        {
+            code: [
+                "export default function foo() {",
+                "   console.log('test');",
+                "}"
+            ].join("\n"),
+            options: [{max: 1}],
+            parserOptions: {ecmaVersion: 6, sourceType: "module"}
+        },
+        {
+            code: "export let foo = 0;",
+            options: [{max: 1}],
+            parserOptions: {ecmaVersion: 6, sourceType: "module"}
+        },
+        {
+            code: [
+                "export function foo() {",
+                "   console.log('test');",
+                "}"
+            ].join("\n"),
+            options: [{max: 1}],
+            parserOptions: {ecmaVersion: 6, sourceType: "module"}
         }
     ],
     invalid: [
@@ -128,6 +156,8 @@ ruleTester.run("max-statements-per-line", rule, {
         { code: "[bar => { a; }, baz => { b; }, qux => { c; }, quux => { d; }];", options: [{ max: 4 }], parserOptions: { ecmaVersion: 6 }, errors: [{ message: "This line has too many statements. Maximum allowed is 4." }] },
         { code: "foo(bar => { a; }, baz => { b; }, qux => { c; }, quux => { d; });", options: [{ max: 4 }], parserOptions: { ecmaVersion: 6 }, errors: [{ message: "This line has too many statements. Maximum allowed is 4." }] },
         { code: "({ bar: bar => { a; }, baz: baz => { b; }, qux: qux => { c; }, quux: quux => { d; }});", options: [{ max: 4 }], parserOptions: { ecmaVersion: 6 }, errors: [{ message: "This line has too many statements. Maximum allowed is 4." }] },
-        { code: "a; if (b) { c; d; }\nz;", options: [{ max: 2 }], errors: [{ message: "This line has too many statements. Maximum allowed is 2." }] }
+        { code: "a; if (b) { c; d; }\nz;", options: [{ max: 2 }], errors: [{ message: "This line has too many statements. Maximum allowed is 2." }] },
+        { code: "export default function foo() { console.log('test') }", options: [{ max: 1 }], parserOptions: {ecmaVersion: 6, sourceType: "module"}, errors: [{ message: "This line has too many statements. Maximum allowed is 1." }] },
+        { code: "export function foo() { console.log('test') }", options: [{ max: 1 }], parserOptions: {ecmaVersion: 6, sourceType: "module"}, errors: [{ message: "This line has too many statements. Maximum allowed is 1." }] }
     ]
 });

